### PR TITLE
Refactor: simplify main by moving logic to start informers

### PR DESCRIPTION
**Pull request by jlpadilla**
### Related Issue
N/A
### Description of changes
- Refactor to simplify main by moving logic to start informers to a separate file.
- Better implementation comments.

### CODE REVIEW

File: pkg/informer/informer.go [hunk 1]
```diff
@@ -105,7 +105,8 @@ func (inform *GenericInformer) listAndResync() error {
 	// Keep track of new resources added to consolidate against the previous state.
 	newResourceIndex := make(map[string]string)
 
-	// List resources.
+	// We need this limit to avoid a memory spike. Smaller chunks allows us to release memory faster, however
+	// it generates more requests to he kube api server.
```
Comment (SherinV) [inline pkg/informer/informer.go:109]:
```suggestion
	// it generates more requests to the kube api server.
```

**Review by SherinV — APPROVED** [review]

File: pkg/informer/informer.go (hunk 1 above)
Comment (jlpadilla) [reply pkg/informer/informer.go]: done

**Comment by jlpadilla** [comment]
/override sonar-pre-submit
The code moved doesn't have existing tests. This refactor will help us improve testing of this path in the near future but I'm separating it to avoid the "snowball effect" and complexity of too many changes at once.

**Comment by jlpadilla** [comment]
/override ci/prow/sonar-pre-submit
The code moved doesn't have existing tests. This refactor will help us improve testing of this path in the near future but I'm separating it to avoid the "snowball effect" and complexity of too many changes at once.

**Comment by jlpadilla** [comment]
/override ci/prow/sonar-pre-submit
The code moved doesn't have existing tests. This refactor will help us improve testing of this path in the near future but I'm separating it to avoid the "snowball effect" and complexity of too many changes at once.

**Pull request merged**